src/ast/passes/semantic_analyser.cpp

```diff
@@ -930,9 +932,10 @@ void SemanticAnalyser::visit(Call &call)
       if (arg.is_map) {
         Map &map = static_cast<Map &>(arg);
         if (map.vargs != nullptr) {
-          LOG(ERROR, call.loc, err_)
-              << "The map passed to " << call.func << "() should not be "
-              << "indexed by a key";
+          if (call.vargs->size() != 1)
+            LOG(ERROR, call.loc, err_)
+                << "Single-value (i.e. indexed) map print cannot take top "
+                   "or div arguments.";
```

Comment:
I don't think that the err message should be top- and div-specific. We may add more cases requiring multiple `print` arguments in future and would have to always update the message:
```suggestion
                << "Single-value (i.e. indexed) map print cannot take additional "
                   "arguments.";
```

Reply:
Done, thanks!

Reply:
It seems that you changed the wrong error message :-)

Reply:
Whoops...should probably get to bed